### PR TITLE
status: technical support level comes from ua-contract service

### PR DESCRIPTION
**Pull request by blackboxsw**
The backend service returns a 'supportLevel' affordance value that
should be reported by `ua status` as the Techincal support level.

When no supportLevel value is present in the machine token response,
return 'n/a' (status.INAPPLICABLE) from ua client.

Fixes: #486

### CODE REVIEW

**Comment by blackboxsw** [comment]
To test:
make deb
lxc launch ubuntu-daily:bionic b1
lxc file push ubuntu-advantage-tools_all.deb b1/
lxc exec b1 -- dpkg -i /ubuntu-advantage-tools_all.deb
lxc exec b1 ua attach
lxc exec b1 ua status

**Comment by powersj** [comment]
@blackboxsw can you resolve the conflicts please

**Comment by blackboxsw** [comment]
Unblocking this per UA checkin meeting. We will allow this behavior in client and expect minor salesforce tickets for corrections on individual contracts if customers see discrepancy for invalid supportLevel reported.

File: uaclient/config.py [hunk 1]
```diff
@@ -241,7 +241,7 @@ def write_cache(self, key: str, content: 'Any') -> None:
     def _status(self) -> 'Dict[str, Any]':
         """Return configuration status as a dictionary."""
         from uaclient.entitlements import ENTITLEMENT_CLASSES
-        response = DEFAULT_STATUS
+        response = copy.deepcopy(DEFAULT_STATUS)
```
Comment (OddBloke) [inline uaclient/config.py]: There doesn't appear to be a test that covers this change. (Ideally, it would also be in a separate commit.)

File: uaclient/config.py [hunk 2]
```diff
@@ -259,6 +259,11 @@ def _status(self) -> 'Dict[str, Any]':
                 'name': ent.name, 'entitled': contract_status,
                 'status': op_status, 'statusDetails': op_details}
             response['services'].append(service_status)
+        support = self.entitlements.get('support', {}).get('entitlement')
+        if support:
+            response['techSupportLevel'] = support.get(
+                'affordances', {}).get('supportLevel', status.INAPPLICABLE)
```
Comment (OddBloke) [inline uaclient/config.py]: This leaves us with `status.INAPPLICABLE` as the default in two different places (here and in `DEFAULT_STATUS`); can we refactor things so that if/when this default changes (looking at you #501) we don't have to ensure that both are kept in sync?

**Comment by OddBloke** [comment]
Oh, another thought: does this mean we can get rid of some of the constants in status.py?

File: uaclient/config.py (hunk 1 above)
Comment (blackboxsw) [reply uaclient/config.py]: added an idempotency check to an existing unit test

File: uaclient/config.py (hunk 2 above)
Comment (blackboxsw) [reply uaclient/config.py]: Using the DEFAULT_STATUS['techicalSupportLevel'] we have defined... and will grab #501 next

File: uaclient/tests/test_config.py [hunk 3]
```diff
@@ -306,6 +310,7 @@ def test_nonroot_attached(self, _m_getuid):
             'techSupportLevel': status.INAPPLICABLE,
         }
         assert expected == cfg.status()
+        # cfg.status is idempotent
```
Comment (OddBloke) [inline uaclient/tests/test_config.py]: This addition seems more aspirational than functional. Did you accidentally miss a line? :-P

Comment (blackboxsw) [reply uaclient/tests/test_config.py]: This is showing up in my local branch per commitish 60744219dd54fc413cdb82b1c6cb1dccfee86d22
+        # cfg.status() idempotent
+        assert expected == cfg.status()
